Gems/ROS2/Code/Source/RobotImporter/SDFormat/Hooks/ROS2SkidSteeringModelHook.cpp

```diff
@@ -0,0 +1,201 @@
+                configuration.m_wheelbase = element->Get<float>("wheelSeparation");
+
+                { // ROS 1 version libgazebo_ros_skid_steer_drive.so
+                    if (element->HasElement("leftJoint") && element->HasElement("rightJoint"))
+                    {
+                        addAxle(
+                            element->Get<std::string>("leftJoint"),
+                            element->Get<std::string>("rightJoint"),
+                            element->Get<float>("wheelDiameter"),
+                            "");
+                    }
+                }
+
+                { // ROS 1 version libgazebo_ros_diff_drive.so
+                    if (element->HasElement("leftFrontJoint") && element->HasElement("rightFrontJoint"))
+                    {
+                        addAxle(
+                            element->Get<std::string>("leftFrontJoint"),
+                            element->Get<std::string>("rightFrontJoint"),
+                            element->Get<float>("wheelDiameter"),
+                            "Front");
+                    }
+                    if (element->HasElement("leftRearJoint") && element->HasElement("rightRearJoint"))
+                    {
+                        addAxle(
+                            element->Get<std::string>("leftRearJoint"),
+                            element->Get<std::string>("rightRearJoint"),
+                            element->Get<float>("wheelDiameter"),
+                            "Rear");
+                    }
+                }
+            }
+
+            if (element->HasElement("wheel_separation") && element->HasElement("wheel_diameter"))
+            {
+                // ROS 2 version of either libgazebo_ros_skid_steer_drive.so or libgazebo_ros_diff_drive.so
+                const auto wheelPairs = element->Get<int>("num_wheel_pairs", 1).first;
+                int dataCount = 0;
+
+                auto wheelSeparation = element->GetElement("wheel_separation");
+                auto wheelDiameter = element->GetElement("wheel_diameter");
+                auto jointLeft = element->GetElement("left_joint");
+                auto jointRight = element->GetElement("right_joint");
+                while (jointLeft != nullptr && jointRight != nullptr && wheelSeparation != nullptr && wheelDiameter != nullptr)
+                {
+                    dataCount++;
+                    if (dataCount == 1)
+                    {
+                        // switch into configuration.m_track https://github.com/o3de/o3de-extras/issues/615
+                        configuration.m_wheelbase = wheelSeparation->Get<float>();
+                    }
+                    else
+                    {
+                        constexpr float epsilon = 0.001f;
+                        AZ_Warning(
+                            "CreateVehicleConfiguration",
+                            fabsf(configuration.m_wheelbase - wheelSeparation->Get<float>()) < epsilon,
```

Comment:
FYI: There is an AZCore math function that does this as well (you'll need to include <AzCore/Math/MathUtils.h>

```suggestion
                            AZ::IsClose(configuration.m_wheelbase, wheelSeparation->Get<float>(), epsilon),
```

Reply:
Pushed in https://github.com/o3de/o3de-extras/pull/642